packages/kbn-unified-data-table/src/components/compare_documents/compare_documents.tsx

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import {
+  EuiDataGrid,
+  EuiDataGridColumnVisibility,
+  EuiDataGridInMemory,
+  EuiDataGridProps,
+  EuiDataGridRowHeightsOptions,
+  EuiDataGridSchemaDetector,
+  EuiDataGridStyle,
+  EuiDataGridToolBarVisibilityOptions,
+  useGeneratedHtmlId,
+} from '@elastic/eui';
+import type { DataView } from '@kbn/data-views-plugin/common';
+import type { DataTableRecord } from '@kbn/discover-utils/types';
+import type { FieldFormatsStart } from '@kbn/field-formats-plugin/public';
+import React, { useMemo } from 'react';
+import useLocalStorage from 'react-use/lib/useLocalStorage';
+import { GRID_STYLE } from '../../constants';
+import { ComparisonControls } from './comparison_controls';
+import { renderComparisonToolbar } from './comparison_toolbar';
+import { useComparisonCellValue } from './hooks/use_comparison_cell_value';
+import { useComparisonColumns } from './hooks/use_comparison_columns';
+import { useComparisonCss } from './hooks/use_comparison_css';
+import { useComparisonFields } from './hooks/use_comparison_fields';
+import type { DocumentDiffMode } from './types';
+
+export interface CompareDocumentsProps {
+  id: string;
+  wrapper: HTMLElement | null;
+  consumer: string;
+  ariaDescribedBy: string;
+  ariaLabelledBy: string;
+  dataView: DataView;
+  isPlainRecord: boolean;
+  selectedFieldNames: string[];
+  selectedDocs: string[];
+  schemaDetectors: EuiDataGridSchemaDetector[];
+  forceShowAllFields: boolean;
+  showFullScreenButton?: boolean;
+  fieldFormats: FieldFormatsStart;
+  getDocById: (id: string) => DataTableRecord | undefined;
+  setSelectedDocs: (selectedDocs: string[]) => void;
+  setIsCompareActive: (isCompareActive: boolean) => void;
+}
+
+const COMPARISON_ROW_HEIGHT: EuiDataGridRowHeightsOptions = { defaultHeight: 'auto' };
+const COMPARISON_IN_MEMORY: EuiDataGridInMemory = { level: 'sorting' };
+const COMPARISON_GRID_STYLE: EuiDataGridStyle = { ...GRID_STYLE, stripes: undefined };
+
+const getStorageKey = (consumer: string, key: string) => `${consumer}:dataGridComparison${key}`;
+
+const CompareDocuments = ({
+  id,
+  wrapper,
+  consumer,
+  ariaDescribedBy,
+  ariaLabelledBy,
+  dataView,
+  isPlainRecord,
+  selectedFieldNames,
+  selectedDocs,
+  schemaDetectors,
+  forceShowAllFields,
+  showFullScreenButton,
+  fieldFormats,
+  getDocById,
+  setSelectedDocs,
+  setIsCompareActive,
+}: CompareDocumentsProps) => {
+  const [showDiff, setShowDiff] = useLocalStorage(getStorageKey(consumer, 'ShowDiff'), true);
```

Comment:
I think we should keep this in local storage to be consistent with the other settings. In most cases users probably won't have a reason to disable diffing, but there are some legitimate use cases such as just wanting to view multiple documents in a vertical format similar to the doc viewer, or wanting to retain the original colours when viewing documents that use the colour field formatter instead of having them overwritten by the diff colours. This way they can exist comparison mode, select more documents, and return to comparison mode without having to keep disabling "Show diff".

No diff:
<img width="1997" alt="no_diff" src="https://github.com/elastic/kibana/assets/25592674/e5c0f12c-b47f-4734-aa39-b3496d3c0c21">

Diff:
<img width="1997" alt="diff" src="https://github.com/elastic/kibana/assets/25592674/55c157a1-2058-4300-b0bb-3b7a7ac99ca1">